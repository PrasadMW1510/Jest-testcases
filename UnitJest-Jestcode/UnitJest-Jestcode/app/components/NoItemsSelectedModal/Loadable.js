/**
 *
 * Asynchronously loads the component for NoItemsSelectedModal
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
