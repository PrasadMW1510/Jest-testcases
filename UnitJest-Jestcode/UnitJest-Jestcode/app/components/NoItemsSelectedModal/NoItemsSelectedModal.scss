.no-items-selected-modal__message {
  font-size: 11px;
  padding: 20px 10px 0 30px;
}

.no-items-selected-modal-success__ok {
  color: #fff;
  border-radius: 0;
  cursor: pointer;
  font-size: 12px;
  padding: 4px 4px 5px;
}

.no-items-selected-modal-success {
  height: 100px;
  padding: 0;
  width: 175px;
}

.no-items-selected-modal-success__message {
  font-size: 11px;
  padding: 30px 20px 0 45px;
}

.no-items-selected-modal-success__button {
  margin-top: 5px;
  background-color: #d06021;
  border-left: 2px solid hsla(0, 0%, 100%, 0.5);
  border-top: 2px solid hsla(0, 0%, 100%, 0.5);
  width: 60px;
  text-align: center;
  margin-left: 60px;
}



