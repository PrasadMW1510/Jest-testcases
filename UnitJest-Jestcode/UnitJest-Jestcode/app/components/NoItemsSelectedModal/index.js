/**
 *
 * NoItemsSelectedModal
 *
 */

export { default } from './NoItemsSelectedModal';
