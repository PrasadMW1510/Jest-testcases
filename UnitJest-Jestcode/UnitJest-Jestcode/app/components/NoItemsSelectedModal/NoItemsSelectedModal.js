/**
 * NoItemsSelectedModal
 */

import React from 'react';
import PropTypes from 'prop-types';
import SAMModal from 'components/SAMModal';

import './NoItemsSelectedModal.scss';

function NoItemsSelectedModal(props) {
  function renderMessage() {
    const type = props.data.type;
    const message = (
      <div className="no-items-selected-modal-success__message">No {type} selected.</div>
    );
    return message;
  }

  return (
    <SAMModal
      isOpen={props.isOpen}
      contentLabel=""
      modalClassModifier="no-items-selected-modal-success"
    >
      {renderMessage()}
      <div className="no-items-selected-modal-success__button">
        <button className="no-items-selected-modal-success__ok" onClick={props.onOK}>
          OK
        </button>
      </div>
    </SAMModal>
  );
}

NoItemsSelectedModal.propTypes = {
  isOpen: PropTypes.bool,
  data: PropTypes.object.isRequired,
  onOK: PropTypes.func.isRequired,
};

export default NoItemsSelectedModal;
