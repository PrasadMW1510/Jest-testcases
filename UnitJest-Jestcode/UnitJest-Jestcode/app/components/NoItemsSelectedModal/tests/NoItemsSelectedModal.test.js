import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import NoItemsSelectedModal from '../index';

describe('<NoItemsSelectedModal />', () => {
  let wrapper = null;

  let mockData = null;
  let mockOnOK = null;

  beforeEach(() => {
    mockData = { type: 'mockType' };
    mockOnOK = jest.fn();

    wrapper = shallow(<NoItemsSelectedModal data={mockData} onOK={mockOnOK} />);
  });

  it('expect to render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
