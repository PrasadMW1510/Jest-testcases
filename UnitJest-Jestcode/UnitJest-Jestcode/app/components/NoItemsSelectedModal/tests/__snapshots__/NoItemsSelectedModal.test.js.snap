// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoItemsSelectedModal /> expect to render correctly 1`] = `
<SAMModal
  contentLabel=""
  modalClassModifier="no-items-selected-modal-success"
>
  <div
    className="no-items-selected-modal-success__message"
  >
    No 
    mockType
     selected.
  </div>
  <div
    className="no-items-selected-modal-success__button"
  >
    <button
      className="no-items-selected-modal-success__ok"
      onClick={[Function]}
    >
      OK
    </button>
  </div>
</SAMModal>
`;
